### PR TITLE
feat: resolve passthrough package.json files

### PR DESCRIPTION
**Pull request by calebboyd**
Previously, a re-directing package.json was ignored by the resolver. 
Now, package.json files are globbed from the root of any resolved packages, avoiding nested node_modules folders.

_Re-Directing package.json file:_
- A package.json that has an export that redirects to a directory with another package.json
    ```json
    "exports": { 
        "." "./somedir_with_different_package.json/file.js" 
    }
    ```
Previously Detected (final package.json file):
	- `./somedir_with_different_package.json/package.json`

Now Detected (all/intermediate package.json files:
    -  `./somedir_with_different_package.json/package.json`
    - `./package.json`

### CODE REVIEW

**Review by bruce-one — APPROVED** [review]
I'm not fully across resolve-dependencies, but it reads as sane to me :-)

I did have a small query or two; and did wonder if an additional test would make sense for the exports case explicitly? The case described in the PR didn't feel like it was as explicitly tested as it could be - but that could just be my lack of understanding :-)

File: src/node-loader.ts [hunk 1]
```diff
@@ -141,12 +143,19 @@ export function load(
...
+    const pkgRoot = join(pkgRootPrefix.join(ModulesDir), ModulesDir, rootPkgDir)
+
     if (expandVariable || expandAll) {
       expand(file, fileDir, pkgDir, nodeModuleGlobs(file))
       file.contextExpanded = true
     }
     if (extraGlobs(file).length) {
-      expand(file, fileDir, pkgDir, extraGlobs(file))
+      expand(file, fileDir, pkgRoot, extraGlobs(file))
     }
```
[the remark this answers is not in the document]
Comment (calebboyd) [reply src/node-loader.ts:159]: Nice spot!

File: src/resolve.spec.ts [hunk 2]
```diff
@@ -167,18 +191,24 @@ describe('resolve-dependencies', () => {
     })
 
     it('should resolve *all* package files when expand: all', () => {
-      result = resolveFilesSync('./app.js', { cwd: cwdCjs, expand: 'all' })
+      result = resolveFilesSync('./app.js', { cwd: fixtureDir, expand: 'all' })
       const allFiles = Object.values({ ...unreferencedFiles, ...referencedFiles }).sort()
       expect(Object.keys(result.files).sort()).toEqual(allFiles)
```
[the remark this answers is not in the document]
Comment (calebboyd) [reply src/resolve.spec.ts:197]: Haha, yeah, I shouldn't have indexed that file there! Fixed.

**Comment by bruce-one** [comment]
Was just curious, is there a reason that this repo doesn't have an automated CI job on it? :-)

**Comment by calebboyd** [comment]
GitHub actions hadn't been released yet back when I first put this together 😅. The initial (commercial) project I where I used nexe had its own CI/CD...

But no reason not to add it now 🎉  

https://github.com/nexe/resolve-dependencies/pull/21
https://github.com/nexe/resolve-dependencies/actions/runs/4629105141

**Pull request merged**